backend/src/main/java/gov/cdc/usds/simplereport/api/Translators.java

```diff
@@ -2,58 +2,199 @@
+            throw new IllegalGraphqlArgumentException("[" + d + "] is not a valid date.");
+        }
+    }
+
+    public static final LocalDate parseUserDate(String userSuppliedDateString) {
+        if (userSuppliedDateString == null) {
+            return null;
+        }
+        if (userSuppliedDateString.contains("/")) {
+            return LocalDate.parse(userSuppliedDateString, US_SLASHDATE_FORMATTER);
+        } else {
+            return LocalDate.parse(userSuppliedDateString); // ISO_LOCAL_DATE is the default
+        }
+    }
+
+    public static final Date parseUserDateTime(String userSuppliedIsoDateString) {
+        if (userSuppliedIsoDateString == null) {
+            return null;
+        }
+        Instant isoDateInstant;
+        try {
+              isoDateInstant = Instant.parse(userSuppliedIsoDateString);
+        }
+        catch (DateTimeParseException parseException) {
+            throw new IllegalGraphqlArgumentException("[" + userSuppliedIsoDateString + "] is not a valid date.");
+        }
+          return Date.from(isoDateInstant);
+    }
+
+    public static String parsePhoneNumber(String userSuppliedPhoneNumber) {
+        if (userSuppliedPhoneNumber == null) {
+            return null;
+        }
+
+        try {
+            var phoneUtil = PhoneNumberUtil.getInstance();
+            return phoneUtil.format(phoneUtil.parse(userSuppliedPhoneNumber, "US"),
+                    PhoneNumberUtil.PhoneNumberFormat.NATIONAL);
+        } catch (NumberParseException parseException) {
+            throw new IllegalGraphqlArgumentException("[" + userSuppliedPhoneNumber + "] is not a valid phone number.");
+        }
+    }
+
+    public static String parseString(String value) {
+        return value == null || "".equals(value) ? null : value.trim();
+    }
+
+    public static UUID parseUUID(String uuid) {
+        if (uuid == null || "".equals(uuid)) {
+            return null;
+        }
+        try {
+            return UUID.fromString(uuid);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalGraphqlArgumentException("\""+uuid+"\" is not a valid UUID.");
+        }
+    }
+
+    public static PersonRole parsePersonRole(String r) {
+        String role = parseString(r);
+        if (role == null) {
+            return PersonRole.UNKNOWN;
+        } 
+        try {
+            return PersonRole.valueOf(role.toUpperCase());
+        } catch (IllegalArgumentException e){
+            throw new IllegalGraphqlArgumentException("\""+r+"\" is not a valid role.");
+        }
+    }
+
+    public static String parseEmail(String e) {
+        String email = parseString(e);
+        if (email == null) {
+            return null;
+        } 
+        if (email.contains("@")) {
+            return email;
+        }
+        throw new IllegalGraphqlArgumentException("\""+e+"\" is not a valid email.");
+    }
+
+    private static final Set<String> RACES = Set.of(
```

Comment:
For these enumerations we'll end up having them defined and mapped in [multiple places](https://github.com/CDCgov/prime-simplereport/search?q=pacific) which is going to make maintenance and modification unfun. If someone is going to redo the app in the future perhaps it doesn't matter.

Reply:
Fully agreed. I think the long term goal is for the data hub to support an api of the accepted values and for our system to use them as the source of truth